packages/apollo/src/utils/lists.ts

```diff
@@ -86,7 +86,19 @@ export class ListHelper {
             return undefined
         }
 
-        return existing.slice(skip, skip + first)
+        const currentList = existing.slice(skip, skip + first)
+
+        const cache = options?.cache.extract()
+
+        const hasBrokenLinks = currentList.filter(Boolean).map((link) => {
+            const ref = get(link, ['__ref'], null)
+            if (cache && ref && !cache[ref]) return null
```

Comment:
should we check for the presence of the key or just the value?
```suggestion
            if (cache && ref && !(ref in cache)) return null
```